lib/affiliate_admin_constraint.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+require 'authlogic/controller_adapters/sinatra_adapter'
+
+class AffiliateAdminConstraint
+  def self.matches?(request)
+    Authlogic::Session::Base.controller =
+      Authlogic::ControllerAdapters::SinatraAdapter::Adapter.new(request)
+    user_session = UserSession.find
+    user_session&.user&.is_affiliate_admin?
+  end
+end
```

Comment:
This is the meat of this PR. For a request to `admin/resque/*`, this constraint checks that:
- a user session exists
- the session is associated with an `approved` user (done under the hood by Authlogic)
- the user is a super admin (`is_affiliate_admin?`)